include/oneapi/dpl/pstl/tuple_impl.h

```diff
@@ -793,6 +793,25 @@ struct __decay_with_tuple_specialization<::std::tuple<_Args...>>
 template <typename... _Args>
 using __decay_with_tuple_specialization_t = typename __decay_with_tuple_specialization<_Args...>::type;
 
+// Flatten nested std::tuple or oneapi::dpl::__internal::tuple types into a single std::tuple.
+template <typename _T>
+struct __flatten_std_or_internal_tuple
```

Comment:
I'd recommend moving `__flatten_std_or_internal_tuple` into `utils.h`. It is a niche utility not related to the core part of the class.

Reply:
This was originally done since `tuple_impl.h` includes `utils.h`, so we would have to forward declare our internal tuple otherwise to avoid a circular dependency.

The new utility supports on arbitrary type and doesn't require any specializations for our internal tuple, so it can be easily placed in `utils.h`